doc/contribute/build_ccf.rst

```diff
@@ -102,4 +102,6 @@ or if there are no Doxygen changes
 
 .. code-block:: bash
 
-    $ SKIP_DOXYGEN=ON ./livehtml.sh
\ No newline at end of file
+    $ SKIP_DOXYGEN=ON ./livehtml.sh
+
+.. note:: If you haven't already built CCF then to build local documentation you may need to temporarily rename ``python/version.py.in`` to ``python/version.py`` and edit it to contain a valid version.  
```

Comment:
Good spot! 

The `version.py.in` file is an input file for `cmake` which automatically converts it to `version.py` [here](https://github.com/microsoft/CCF/blob/main/CMakeLists.txt#L49-L51) so that the version of the CCF Python package is automatically deduced from the git commit (which is done [here](https://github.com/microsoft/CCF/blob/main/cmake/version.cmake) if you're curious), so that we don't have to manually update the version every time we want to release a new version of CCF (which happens [often enough](https://github.com/microsoft/CCF/releases)!)

We've always (wrongly) assumed that people build the docs after having built CCF (see https://microsoft.github.io/CCF/main/contribute/build_ccf.html) and thus have a `version.py` script automatically generated under `python/`. In fact, this is what our CI does [here](https://github.com/microsoft/CCF/blob/main/.azure-pipelines-gh-pages.yml#L26-L29). 

In this case, because we've written `livehtml.sh` too, I think it's preferable to have the script generate `version.py` via `cmake` if this one doesn't already exist. Something like:

```bash
mkdir -p build_docs
cd build_docs
cmake .. # Generates version.py
cd ..
rm -rf build_docs
```

Reply:
Thanks, good idea! I've now updated `livehtml.sh` to generate `version.py` if it doesn't already exist, and I've removed the note I had added to `doc/contribute/build_ccf.rst` about it.